src/rest_vol.c

```diff
@@ -2597,17 +2602,21 @@ RV_parse_creation_properties_callback(yajl_val parse_tree, char **GCPL_buf)
     if (NULL == (parsed_string = YAJL_GET_STRING(key_obj)))
         FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "creationProperties was NULL");
 
-    if (NULL == (*GCPL_buf = RV_malloc(strlen(parsed_string) + 1)))
+    if (NULL == (GCPL_buf_local = RV_malloc(strlen(parsed_string) + 1)))
         FUNC_GOTO_ERROR(H5E_OBJECT, H5E_CANTALLOC, FAIL, "failed to allocate memory for creationProperties");
 
-    if (NULL == (memcpy(*GCPL_buf, parsed_string, strlen(parsed_string) + 1)))
+    if (NULL == (memcpy(GCPL_buf_local, parsed_string, strlen(parsed_string) + 1)))
         FUNC_GOTO_ERROR(H5E_OBJECT, H5E_SYSERRSTR, FAIL, "failed to copy creationProperties");
 
+    if (NULL == (*GCPL_buf_out = GCPL_buf_local))
```

Comment:
No need to check for NULL here, just assign `*GCPL_buf_out` to `GCPL_buf_local`. Though, this function should probably have a check that `GCPL_buf_out` isn't NULL to prevent a possible segfault.